File: backend/internal/domain/task_category.go
package domain

import (
	"context"
	"errors"
)

type TaskCategory struct {
	ID    string `json:"id"`
	Title string `json:"title"`
}

type TaskCategoryRepository interface {
	CreateTaskCategory(ctx context.Context, taskCT *TaskCategory) error
	List(ctx context.Context) ([]TaskCategory, error)
	Delete(ctx context.Context, id string) error
	FindById(ctx context.Context, id string) (*TaskCategory, error)
}

type TaskCategoryUsecase interface {
	CreateTaskCategory(ctx context.Context, taskCT *TaskCategory) error
	List(ctx context.Context) ([]TaskCategory, error)
	Delete(ctx context.Context, id string) error
	FindById(ctx context.Context, id string) (*TaskCategory, error)
}

var (
	ERR_TASK_CATEGORY_NOT_FOUND = errors.New("taskCategory not found ")
)

File: webapp/src/features/tasks_category/domain/task_category.ts
export interface TaskCategory {
  id?: string;
  title: string;
}

export interface ITaskCategoryRepository {
  CreateTaskCategory(taskC: TaskCategory): Promise<boolean>;
  List(): Promise<TaskCategory[]>;
  Delete(id: string): Promise<boolean>;
  FindById(id: string): Promise<TaskCategory>;
}

export interface ITaskCategoryUsecase {
  CreateTaskCategory(taskC: TaskCategory): Promise<boolean>;
  List(): Promise<TaskCategory[]>;
  Delete(id: string): Promise<boolean>;
  FindById(id: string): Promise<TaskCategory | null>;
}
